### ISSUE TITLE
cc2538 FLASH settings location

### ISSUE THREAD

**Issue by DuaneEllis-TI** [openthread/openthread#2069]
Probably for some historic reason, the location of the CC2538 FLASH settings is in the middle of the flash at address 0x00203900 - which if you turn on features, and turn the logs up the flash grows and then the two areas (code/text) vrs (settings) overlap and things go horribly wrong.

Most typically the initialized data goes bad at startup because the ".data" initializes are are stored at the END of the .text section, and thus these are the first to get wiped out, or partially wiped out depending upon how they align verses the page boundary. 

We have a fix for this but are not sure about the origins of the "0x39000" magic number in settings.cpp

Basically we move the SETTINGS storage to the *LAST* two pages  *END* of physical flash, just before the "CCA"  (customer configuration area) where FACTORY settings are normally stored)
thus the last 3 page are set aside as follows:

(N-1) = CCA area, length = 2k
(N-2) = Thread Settings Storage Area (B)
(N-3) = Thread Settings Storage Area (A)
